apps/hash-ai-worker-ts/src/activities/flow-activities/answer-question-action.ts

```diff
@@ -0,0 +1,448 @@
+import { getSimpleGraph } from "@local/hash-backend-utils/simplified-graph";
+import type { GraphApi } from "@local/hash-graph-client";
+import { getSimplifiedActionInputs } from "@local/hash-isomorphic-utils/flows/action-definitions";
+import type { StepOutput } from "@local/hash-isomorphic-utils/flows/types";
+import {
+  currentTimeInstantTemporalAxes,
+  zeroedGraphResolveDepths,
+} from "@local/hash-isomorphic-utils/graph-queries";
+import { mapGraphApiSubgraphToSubgraph } from "@local/hash-isomorphic-utils/subgraph-mapping";
+import type { EntityRootType } from "@local/hash-subgraph";
+import { extractEntityUuidFromEntityId } from "@local/hash-subgraph";
+import type { Status } from "@local/status";
+import { StatusCode } from "@local/status";
+import { Context } from "@temporalio/activity";
+import dedent from "dedent";
+import { CodeInterpreter, Sandbox } from "e2b";
+import type OpenAI from "openai/index";
+import type { ChatCompletionToolMessageParam } from "openai/src/resources/chat/completions";
+
+import { logger } from "../../shared/logger";
+import { getOpenAiResponse } from "../shared/openai";
+import { stringify } from "../shared/stringify";
+import type { FlowActionActivity } from "./types";
+
+const answerTools: OpenAI.Chat.Completions.ChatCompletionTool[] = [
+  {
+    type: "function",
+    function: {
+      name: "answer",
+      description:
+        "Submit the answer, or an explanation of why the question cannot be answered using the available data",
```

Comment:
I wonder if it'd be better to have a separate `"terminate"` tool call here, so it's easier for us to determine if the agent gave up, or is trying to submit a valid answer.

Reply:
It might be, yes – currently just relying on the absence of an answer, but it might give one even if it's giving up. 

I'd like to experiment with this more in follow-up so we can get it in without for now